pkg/operator/controller/garden/reconciler_reconcile.go

```diff
@@ -288,6 +305,24 @@ func (r *Reconciler) reconcile(
 				DoIf(helper.GetETCDEncryptionKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationCompleting),
 			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient),
 		})
+		_ = g.Add(flow.Task{
+			Name: "Creating new ServiceAccount secrets after creation of new signing key",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return secretsrotation.CreateNewServiceAccountSecrets(ctx, log, virtualClusterClient, secretsManager)
+			}).
+				RetryUntilTimeout(30*time.Second, 10*time.Minute).
+				DoIf(helper.GetServiceAccountKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationPreparing),
+			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient),
+		})
+		_ = g.Add(flow.Task{
+			Name: "Deleting old ServiceAccount secrets after rotation of signing key",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				return secretsrotation.DeleteOldServiceAccountSecrets(ctx, log, virtualClusterClient, garden.Status.Credentials.Rotation.ServiceAccountKey.LastInitiationFinishedTime.Time)
+			}).
+				RetryUntilTimeout(30*time.Second, 10*time.Minute).
+				DoIf(helper.GetServiceAccountKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationCompleting),
+			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient),
+		})
```

Comment:
Why did you add this? Can you explain how it is related to this PR? There is no `kube-controller-manager` yet which could take over the needed actions.

Reply:
This was an open question we had. However, we forgot to clarify it 😅

Reply:
OK, then let's remove it for now